frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,319 @@
+/* eslint-disable no-undef */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+
+import { connectToPlural } from '../../kinds';
+import { k8sCreate, K8sKind, referenceFor } from '../../module/k8s';
+import { ButtonBar, history, LoadingBox, NameValueEditor, resourceObjPath, RequestSizeInput } from '../utils';
+import { ListDropdown } from '../RBAC/bindings';
+import { RadioInput } from '../radio';
+import { Checkbox } from '../checkbox';
+
+const StorageClassDropdown: React.SFC<ScDropdownProps> = props => {
+
+  const kind = 'StorageClass';
+  const resources = [{ kind }];
+  const { selectedKey, required, namespace } = props;
+  const extraListOption = { kindLabel: kind, name: 'No Storage Class' };
```

Comment:
We could make the placeholder "No storage class" if we think it might be confusing.

Reply:
I have this for 2 reasons: 1) Once they choose a option, there is no way to unselect it so instead they can select the 'No storage class' and 2) In the attach storage, the current mock-ups have a list item in the drop down that says 'create new PVC' so I am also laying the ground work for that.

Reply:
OK, we might consider just using `Dropdown` directly if we are going to be injecting additional options. It's not too hard to call `k8sList` on component mount to get the data and put it into a dropdown. It would also let us handle cases where there are no storage classes specially if we want. (For instance, we could show a message rather than an empty dropdown.)

The way this is implemented, the caller knows too much about the internals of the `ListDropdown` component. The outer page shouldn't know about internal item properties like `kindLabel` for dropdown items.

Reply:
What if I change it to something like :
const extraListOption = "No Storage Class" 
and then in the Listdropdown have :
const extraItem = { kindLabel: kind, name: extraListOption }